src/pages/Lending/Pool/components/Borrow/BorrowSweep.tsx

```diff
@@ -0,0 +1,251 @@
+import { getThorchainFromAddress } from 'state/slices/opportunitiesSlice/resolvers/thorchainsavers/utils'
+import {
+  selectAssetById,
+  selectFeeAssetByChainId,
+  selectPortfolioAccountMetadataByAccountId,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { BorrowRoutePaths } from './types'
+
+type BorrowSweepProps = {
+  collateralAssetId: AssetId
+  collateralAccountId: AccountId
+}
+
+export const BorrowSweep = ({ collateralAssetId, collateralAccountId }: BorrowSweepProps) => {
+  const {
+    state: { wallet },
+  } = useWallet()
+
+  const [isSweepPending, setIsSweepPending] = useState(false)
+  const [fromAddress, setFromAddress] = useState<string | null>(null)
+  const [txId, setTxId] = useState<string | null>(null)
+
+  const history = useHistory()
+  const translate = useTranslate()
+  const collateralAsset = useAppSelector(state => selectAssetById(state, collateralAssetId))
+
+  const handleBack = useCallback(() => {
+    history.push(BorrowRoutePaths.Input)
+  }, [history])
+  const divider = useMemo(() => <Divider />, [])
+
+  const collateralAccountFilter = useMemo(
+    () => ({ accountId: collateralAccountId }),
+    [collateralAccountId],
+  )
+  const collateralAccountMetadata = useAppSelector(state =>
+    selectPortfolioAccountMetadataByAccountId(state, collateralAccountFilter),
+  )
+
+  const getBorrowFromAddress = useCallback(() => {
+    if (!(wallet && collateralAccountMetadata)) return null
+    return getThorchainFromAddress({
+      accountId: collateralAccountId,
+      assetId: collateralAssetId,
+      getPosition: getThorchainLendingPosition,
+      accountMetadata: collateralAccountMetadata,
+      wallet,
+    })
+  }, [wallet, collateralAccountId, collateralAssetId, collateralAccountMetadata])
+
+  useEffect(() => {
+    if (fromAddress) return
+    ;(async () => {
+      const _fromAddress = await getBorrowFromAddress()
+      if (!_fromAddress) return
+      setFromAddress(_fromAddress)
+    })()
+  }, [getBorrowFromAddress, fromAddress])
+
+  const { data: estimatedFeesData, isLoading: isEstimatedFeesDataLoading } =
+    useGetEstimatedFeesQuery({
+      cryptoAmount: '0',
+      assetId: collateralAssetId,
+      to: fromAddress ?? '',
+      sendMax: true,
+      accountId: collateralAccountId,
+      contractAddress: undefined,
+      enabled: true,
+    })
+
+  const handleSweep = useCallback(async () => {
+    if (!wallet) return
+
+    setIsSweepPending(true)
+
+    try {
+      const fromAddress = await getBorrowFromAddress()
+      if (!fromAddress)
+        throw new Error(`Cannot get from address for accountId: ${collateralAccountId}`)
+      if (!estimatedFeesData) throw new Error('Cannot get estimated fees')
+      const sendInput = {
+        accountId: collateralAccountId,
+        to: fromAddress,
+        input: fromAddress,
+        assetId: collateralAssetId,
+        from: '',
+        cryptoAmount: '0',
+        sendMax: true,
```

Comment:
possible for us to add an extra sweep arg to filter out utxos already owned by `fromAddress`, I believe this current logic will include self sends for those utxos already owned by `fromAddress` as it stands (haven't proven this out officially)

Reply:
current logic would consolidate all utxos into a single new utxo which has value depending on what you are sweeping for, but we could save some possible tx fees by omitting already owned utxos

Reply:
As per dm with @gomesalexandre, going to leave this for a follow up